File: Forráskód/backend/Libary/Model/Work/Work.cs
using Libary.Model.User;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Libary.Model.Work
{
    [Table("Work")]
    /// <summary>
    /// Work fő entitás: hirdetett munka, hozzá tartozó jelentkezők, teendők, képek és címkék.
    /// </summary>
    public class Work
    {
        [Key]
        public int Id { get; set; }

        public int AuthorId { get; set; }
        [ForeignKey(nameof(AuthorId))]
        public User.User? Author { get; set; }

        [Required]
        [MaxLength(50)]
        public string TargetAudience { get; set; }

        [Required]
        [MaxLength(150)]
        public string Title { get; set; }

        [Required]
        public string Description { get; set; }

        [Column(TypeName = "decimal(18, 2)")]
        public decimal? BasePrice { get; set; }

        [Required]
        [MaxLength(50)]
        public string Status { get; set; } = "Open";

        public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAtUtc { get; set; }

        public ICollection<WorkApplicant>? Applicants { get; set; } = new List<WorkApplicant>();
        public ICollection<WorkTodo>? Todos { get; set; } = new List<WorkTodo>();
        public ICollection<WorkImage>? Images { get; set; } = new List<WorkImage>();
        public ICollection<WorkTag>? Tags { get; set; } = new List<WorkTag>();

        // API kompatibilitás miatt használt segédmező: címkék egyszerű string listában.
        [NotMapped]
        public List<string>? Cimkek { get; set; }
    }
}

File: Forráskód/frontend/src/types/work.ts
// Rövid felhasználói adatmodell work nézetekhez és listákhoz.
export interface WorkUserSummary {
  id: number;
  vezetekNev?: string;
  keresztNev?: string;
  telefon?: string;
  email?: string;
  telepules?: string;
  roleId?: number;
  imgString?: string;
}

// Backend címke-kapcsoló struktúra (tag link + activity név).
export interface WorkTagLink {
  tag?: {
    activity?: string;
  };
}

// Work lista szűrők kliens oldali paraméterezéshez.
export interface WorkFilters {
  priceMin?: number;
  priceMax?: number;
  createdFrom?: string;
  createdTo?: string;
  targetAudience?: string;
  status?: string[];
}

// Kép metaadat frissítés (kiemelt kép, párosított before/after kapcsolat).
export interface WorkImageMetadataUpdate {
  isShowcase?: boolean;
  relatedImageId?: number | null;
}

// Fő work entitás a részletes oldalakhoz és szerkesztő felületekhez.
export interface Work {
  id?: number;
  authorId?: number;
  author?: WorkUserSummary;
  targetAudience: string;
  title: string;
  description: string;
  basePrice?: number;
  status?: string;
  createdAtUtc?: string;
  updatedAtUtc?: string;
  applicants?: WorkApplicant[];
  todos?: WorkTodo[];
  images?: WorkImage[];
  cimkek?: string[];
  tags?: WorkTagLink[];
  isCurrentUserRelated?: boolean;
}

// Jelentkezés rekord egy adott munkára.
export interface WorkApplicant {
  id?: number;
  workId: number;
  userId: number;
  user?: WorkUserSummary;
  offeredPrice?: number;
  status: string;
  createdAtUtc?: string;
}

// Work-höz tartozó TODO elem (kis feladatlista).
export interface WorkTodo {
  id?: number;
  workId?: number;
  title: string;
  isDone?: boolean;
  doneMessage?: string;
  doneByUserId?: number;
}

// Work képadat rekord galéria/lista megjelenítéshez.
export interface WorkImage {
  id?: number;
  workId: number;
  imageUrl: string;
  isShowcase: boolean;
  uploadedAtUtc?: string;
}

// Kiemelt work rekord (homepage/admin featured lista).
export interface FeaturedWork {
  id?: number;
  workId: number;
  work?: Work;
  featuredAtUtc?: string;
}

// Lapozott work lista válaszmodell az API kompatibilitáshoz.
export interface PaginatedWorks {
  items: Work[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}
